### ISSUE TITLE
FeatureUnhasher does not support an input_type of dict

### ISSUE THREAD

**Issue by manyu90** [TeamHG-Memex/eli5#236]
The current implementation only supports input types of String. It will be nice to have a FeatureUnhasher which accepts Featurehashers of input type dict

**Comment by kmike** [comment]
As I recall, we added FeatureUnhasher mainly to support HashingVectorizer, so we started with 'string'. 

On a first sight, adding input_type='dict' support it a matter of removing the exception, changing the way `_term_counts` attribute is computed, and adding some tests.

I don't have immediate plans to implement this feature, but it looks like a good problem for new contributors, so pull requests are welcome!

**Comment by mohdkashif93** [comment]
Is it okay if I work on this issue, I mean if nobody else is working on this?

**Comment by coderop2** [comment]
@kmike as i was going through tests there are no tests for the function featureunhasher..?

**Comment by kmike** [comment]
@coderop2 right; adding them can be a good first step. It is tested only indirectly, by testing InvertableHashingVectorizer which uses FeatureUnhasher internally.

**Comment by coderop2** [comment]
So first we can include the functionality for input_type dict and then add
tests for both together

**Comment by kmike** [comment]
@coderop2 yes, this works. Alternatively, one can start by adding tests for existing FeatureHasher, to get their feet wet; this would be a smaller change which can be merged separately.
